instrumentation.go

```diff
@@ -463,3 +474,12 @@ func WithResourceAttributes(attrs ...attribute.KeyValue) InstrumentationOption {
 		return c, nil
 	})
 }
+
+// WithLogLevel returns an [InstrumentationOption] that will configure
+// an [Instrumentation] with the logger level visibility defined as inputed.
+func WithLogLevel(level string) InstrumentationOption {
+	return fnOpt(func(ctx context.Context, c instConfig) (instConfig, error) {
```

Comment:
What do you think about parsing the log level here and erroring out if the user has specified something invalid? It would be nice to know for example if I've miss-spelled "debug", instead of silently switching to "info".

Reply:
It looks like [ParseAtomicLevel](https://pkg.go.dev/go.uber.org/zap#ParseAtomicLevel) does some validation if we choose to use it (currently it looks like it is used in `newLogger`)

Reply:
Added the validation, what do you think?

Reply:
Awesome!